### PR TITLE
Fix stop services in docker compose

### PR DESCRIPTION
**Pull request by sko00o**
## What does this PR do?

1. Add a test to demonstrate the issue problem.
2. Add logger in dockerCompose struct to fix the panic problem.

## Why is it important?

I need to restart services in some test cases.

## Related issues

- Relates #917

### CODE REVIEW

**Comment by mdelapenya** [comment]
We are going to merge #910 very soon, which could cause a merge conflict with this one. Will ping you here once that PR is merged, so you can update it 🙏  (or I can do it myself if they are not that severe)

**Comment by sko00o** [comment]
I saw that PR. And I could deal with conflict after that merged.

**Comment by mdelapenya** [comment]
#910 was merged, no merge conflicts :)

File: modules/compose/compose_api_test.go [hunk 1]
```diff
@@ -101,6 +102,42 @@ func TestDockerComposeAPIWithRunServices(t *testing.T) {
 	assert.Contains(t, serviceNames, "nginx")
 }
 
+func TestDockerComposeAPIWithStopServices(t *testing.T) {
+	compose, err := NewDockerComposeWith(
+		WithStackFiles("./testresources/docker-compose-complex.yml"),
```
Comment (mdelapenya) [inline modules/compose/compose_api_test.go:107]: Could you use filepath to build an OS-agnostic path?

Comment (mdelapenya) [reply modules/compose/compose_api_test.go:107]: There is also a constant for the yaml file

Comment (sko00o) [reply modules/compose/compose_api_test.go]: Done. And I rebase onto the latest main branch.

**Review by mdelapenya — APPROVED** [review]
LGTM!

Thanks @sko00o for your contribution. Even though sonarcloud complains on the duplicated code, I'm going to merge it as, for me, in tests, some duplication comes with readability.

I'd like to thank you for your dedication to testcontainers-go! 🚀

**Pull request merged**